lib/jetstream/api/kv.ex

```diff
@@ -0,0 +1,155 @@
+defmodule Jetstream.API.KV do
+  @moduledoc """
+  API for interacting with the Key/Value store functionality in Nats Jetstream.
+
+  Learn about the Key/Value store: https://docs.nats.io/nats-concepts/jetstream/key-value-store
+  """
+  alias Jetstream.API.Stream
+
+  @stream_prefix "KV_"
+  @subject_prefix "$KV."
+  @two_minutes_in_nanoseconds 1_200_000_000
+
+  @doc """
+  Create a new Key/Value bucket. Can include the following options
+
+  * `:history` - How many historic values to keep per key (defaults to 1, max of 64)
+  * `:ttl` - How long to keep values for (in nanoseconds)
+  * `:max_bucket_size` - The max number of bytes the bucket can hold
+  * `:max_value_size` - The max number of bytes a value may be
+  * `:description` - A description for the bucket
+  * `:replicas` - How many replicas of the data to store
+  * `:storage` - Storage backend to use (:file, :memory)
+  * `:placement` - A map with :cluster (required) and :tags (optional)
+
+  ## Examples
+
+     iex>{:ok, info} = Jetstream.API.KV.create_bucket(:gnat, "my_bucket")
+  """
+  @spec create_bucket(conn :: Gnat.t(), bucket_name :: binary(), params :: keyword()) ::
```

Comment:
If we know all the acceptable params I would create a separate type and use it here instead of generic `keyword()`

Reply:
Good call. Added in https://github.com/mmmries/jetstream/pull/53/commits/171c2789a9cc9be002898b705dd848699e466b5d